packages/react-router/modules/matchPath.js

```diff
@@ -30,7 +30,8 @@ const matchPath = (pathname, options = {}) => {
   if (typeof options === 'string')
     options = { path: options }
 
-  const { path, exact = false, strict = false } = options
+  const { exact = false, strict = false } = options
+  const path = options.path || options.from
```

Comment:
It could also just hang out and wait for that `from` path to match and redirect when it does ... hmm...